### ISSUE TITLE
In a Monorepo, if a subpackage has changes but not the root package, the PR Title will not be created according to pull-request-title-pattern

### ISSUE THREAD

**Issue by guenyoobaywa** [googleapis/release-please#2386]
Thanks for stopping by to let us know something could be better!

Please provide the following details.

#### Environment details

  - OS: Sonoma 14.7
  - Node.js version: 20.11.1
  - npm version: 10.8.2
  - `release-please` version: 4

#### Steps to reproduce

  1. Have a Monorepo with subpackages
  2. Have these settings in release-please-config.json:
  ```json
 {
  "include-component-in-tag": true,
  "include-v-in-tag": true,
  "tag-separator": "@",
  "plugins": ["node-workspace", "sentence-case"],
  "release-type": "node",
  "commit-search-depth": 1,
  "group-pull-request-title-pattern": "chore: release ${component} ${version}",
  "pull-request-title-pattern": "chore: release ${component} ${version}",
  "packages": {
    ".": {
      "packageName": "@packageTeam/root",
      "extra-label": "core",
      "exclude-paths": [
        "packages/package-a",
        "packages/package-b",
        "packages/package-c",
        ...
      ],
      "pull-request-title-pattern": "chore: release ${component} ${version}"
    },
    "packages/package-a": {
      "path": "packages/package-a",
      "packageName": "@packageTeam/package-a",
      "extra-label": "react-components",
      "pull-request-title-pattern": "chore: release ${component} ${version}"
    },
    "packages/package-b": {
      "path": "packages/package-b",
      "packageName": "@packageTeam/package-b",
      "extra-label": "web-components",
      "pull-request-title-pattern": "chore: release ${component} ${version}"
    },
    ...
  },
  "$schema": "https://raw.githubusercontent.com/googleapis/release-please/main/schemas/config.json"
}
  ```
3. When a release (via release-please-action) is created, and there are changes in the root package as well, everything works fine. I'll get a PR with the requested PR format.

The problem starts when I only have changes in my subpackage, then my PR title will just plainly be "chore: release (#issueNr)"

When release-please reruns, it stumbles and can't find the correct PR to the release, the autorelease: pending tag will not be removed, no release will be created.

Solution, for now, is to manually adjust every Pull Request's title and squash commit message, which is pretty time consuming and error prone.

Can you guys help? 

Making sure to follow these steps will guarantee the quickest resolution possible.

Thanks!

**Comment by nate-credoai** [comment]
Hi, I recently started using release-please and experienced a lot of nuance with the latest version, 16.15.0. However, I am not able to reproduce your issue with what appears to be a matching config with changes only in the subpackage. The only difference is that my repo is python, not node. I do see you are using a config key `packageName` but it should be `package-name`.  But it appears that `package-name` is ignored for `release-type: node` as documented here: https://github.com/googleapis/release-please/blob/main/docs/manifest-releaser.md?plain=1#L265

Also, you say you are using `release-please` version 4 but that looks more like a `release-please-action` version so it's hard to know which version of `release-please` you are actually using. I would suggest running the CLI in dry-run mode with debug enabled and posting that output if you can reproduce the issue.

**Comment by guenyoobaywa** [comment]
@nate-credoai good catch, I'll have to see if renaming the attribute to package-name will change anything, I do however doubt it because as you stated, it should be (and looks like it is) ignored.

I'm also using the release-please-action via github workflows, sorry for the confusion.

**Comment by guenyoobaywa** [comment]
Still reproducible on 17.6.0, and I can now point at the cause.

**In short, in a monorepo, this only worked when the root package had changes.** In our case though, the root package most often didn't.

${component} and ${version} in group-pull-request-title-pattern are taken only from the root path (.) candidate — src/plugins/merge.ts:


if (candidate.path === '.') { rootRelease = candidate; }
…
rootRelease?.pullRequest.title.component,
rootRelease?.pullRequest.title.version,
No fallback. In a monorepo the root package is usually not part of a release, so both render empty and chore: release ${component} ${version} becomes chore: release.

On merge, buildRelease() can't parse that title — ${version} is a required group in generateMatchPattern, and the chore: release ${branch} fallback needs a branch name — so every strategy logs ✖ Bad pull request title: 'chore: release' and no tag or release is created, while the manifest bump has already merged.

It is deterministic, not intermittent: every grouped release PR without a root-package change gets the empty title. It only looked intermittent for us because we were renaming the PRs by hand before merging.

Repro: manifest config with "." as a package, default separate-pull-requests, the group pattern above, and a feat: commit in a sub-package only.

Workaround: "group-pull-request-title-pattern": "chore: release ${branch}" — the version is then read from the body's <summary>component: version</summary> sections and the release is created (verified on 17.6.0).

Fix suggestion: in Merge, fall back to ${branch} or the sole candidate when there is no root candidate; or run the generated title back through PullRequestTitle.parse before opening the PR and fail instead of emitting a title that can never release. Same family as #2654 and #2712.
